pkg/cmd/infra/deployer/deployer.go

```diff
@@ -131,7 +130,7 @@ func NewDeployer(client kclientset.Interface, images imageclientinternal.Interfa
 		getDeployments: func(namespace, configName string) (*kapi.ReplicationControllerList, error) {
 			return client.Core().ReplicationControllers(namespace).List(metav1.ListOptions{LabelSelector: appsutil.ConfigSelector(configName).String()})
 		},
-		scaler: scaler,
+		scaler: appsutil.NewReplicationControllerV1Scaler(client),
```

Comment:
I'm inclined to say we should stop using `kubectl.Scaler` here and instead use direct scale client on a rc and put that manual bit of code responsible for scaling into the deployer. Although I'm might be wrong.

Reply:
@soltysh can be a follow up... i'm fine with using the scale client directly

Reply:
added to follow up list